[souliss] Souliss Binding initial contribution (#10034)
openhab/openhab-addons: bundles/org.openhab.binding.souliss/README.md

Souliss is an open-source and community driven project, you can use the [wiki](https://github.com/souliss/souliss/wiki) and [Community](https://github.com/souliss/souliss/wiki/Community) to get help and share your results.  

## Prerequisites

The binding requires a deployed network.
As a minimum, you need one Souliss node with Ethernet or WiFi access configured as a [Gateway](https://github.com/souliss/souliss/wiki/Gateway).
A Gateway is a special node that is able to communicate with the user interfaces.
The binding interacts as a user interface for Souliss.

A starting point is the [Souliss wiki](https://github.com/souliss/souliss/wiki).
The best is to start with a single node and connect with SoulissApp.
The code for networking activities of this binding is based on [SoulissApp](https://github.com/souliss/souliss/wiki/SoulissApp) code, so once connected with SoulissApp, you can move to openHAB directly.

You can use SoulissApp and the Souliss binding at the same time, and generally up to five (by default, but can be increased) user interfaces simultaneously.

### Sketches

The easiest way is start with a simple example to control an ON/OFF light (though a relay). 

> You can go to project [Souliss[(https://github.com/souliss/souliss), see a lot of examples sketches: [Souliss examples](https://github.com/souliss/souliss/tree/friariello/examples)

Author: fixed

## Binding Configuration

> This binding does not require any special configuration.

Author: removed

## Discovery

This binding can automatically discover devices. 
First Gateway Node, then Peer Nodes. 

## Supported Things

In Souliss Framework a Typical is one of predefined logic dedicated to smart home devices like lights, heating or antitheft. 

Typical can be one of T11, T12, T13, T14, etc... 

It are defined [here](https://github.com/souliss/souliss/wiki/Typicals).

Typicals match directly with openHAB Thing type.

> |Device type |Typical Code | Thing type | 

Author: formatted (second table it's big so not easy thru a small screen read as text)

# Souliss Binding

[Souliss](http://www.souliss.net/) is a networking framework for Arduino and compatibles boards, and is designed to let you easily build a smart home that is distributed over multiple boards via Ethernet, WiFi, wireless point-to-point and RS485 bus. 

Souliss is an open-source and community driven project, you can use the [wiki](https://github.com/souliss/souliss/wiki) and [Community](https://github.com/souliss/souliss/wiki/Community) to get help and share your results.  

## Prerequisites

> The binding requires a deployed network.  As a minimum, you need one Souliss node with Ethernet or WiFi access configured as a [Gateway](https://github.com/souliss/souliss/wiki/Gateway). A Gateway is a special node that is able to communicate with the user interfaces. The binding interacts as a user interface for Souliss.

Reviewer: ```suggestion
The binding requires a deployed network. 
As a minimum, you need one Souliss node with Ethernet or WiFi access configured as a [Gateway](https://github.com/souliss/souliss/wiki/Gateway). 
A Gateway is a special node that is able to communicate with the user interfaces. 
The binding interacts as a user interface for Souliss.
```
Example of how each sentence now starts on its own line. Please do to the whole file.

Author: fixed

|"|HEALTHY_INTERVAL=35|Interval in seconds to send nodes healthy|
|"|USER_INDEX=70|Generally the default value work good. It must is different from other user interfaces (ex: SoulissApp)|
|"|NODE_INDEX=120|Generally the default value work good. It must is different from other user interfaces (ex: SoulissApp)|
|T11|sleep=5|Set sleep timer in cycles|
|T12|||
|T13|||
|T14|||
|T16|sleep=5|Set sleep timer in cycles|
|T19|sleep=5|Set sleep timer in cycles|
|T1A|||
|T21|||
|T22|||
|T31|||
|T4x|||
|T5x|||
|T6x|||

## Manual Things Configuration

If after discovery your thing is not listed you can add it manually.
You have to choice it from disponible items. Firts gateway, after items!
To configure Gateway you can leave default value on Thing ID and write your value on "IP or Host Name" and "Gateway port".

To configure a typical (items) you have to choice your "Name" and "Location", you have to choice your "Gateway" and insert correct "Thing ID".

Thing ID is [node]-[slot]
For example, if you have two nodes and you want configure a typical on second node at slot seven, you must write 
Thing ID: 

```
2-7
```

## Basic UI and Classic UI

> Examples to configure items in Basic UI and Classic UI

Reviewer: classicUI and paperUI do not exist anymore and BasicUI is getting phased out. Please do not mention them in documentation as it will make the docs outdated when they are removed.

}

Frame {  
       Group item=Divano icon="light"
}

Frame label="Temperature"{	

            Text label="Temperatura e umidità" icon="temperature" {
            Default item=FamilyRoom_Temperature label="Temperatura"
            Default item=FamilyRoom_Humidity label="Umidità"
            Default item=AggiornamentoNodo6 icon="icon16x16"
        
}

Text label="Termostato soggiorno" icon="temperature" {
            Setpoint item=termostatosoggiorno_setpoint step=0.5 minValue=10 maxValue=30
            Default item=termostatosoggiorno_temperatura
            Default item=termostatosoggiorno_umidita
            Switch item=termostatosoggiorno_setasmeasured mappings=[ON="Set"]
            Switch item=termostatosoggiorno_modo label="Heating Mode" mappings=[HEATING_MODE="Set"] 
            Switch item=termostatosoggiorno_power label="Power On/Off"
            Default item=termostatosoggiorno_fire label="Fire"
	        Text item=termostatoSoggiorno_aggiornamento label="Aggiornato: [%1$td.%1$tm.%1$tY %1$tk:%1$tM:%1$tS]" icon="icon16x16"
            Default item=termostatoSoggiorno_healthy
	        Slider item=termostatoSoggiorno_displayBright
	}		
}

}
```

## Community

> Souliss is a small community and actualy it don't have sufficient human resource to be more active on Openhab official community

Reviewer: ```suggestion
Souliss is a small community and doesn't have sufficient human resources to be more active on openHAB official community.
```

These are some very popular forum:

English Group, [here](https://groups.google.com/forum/#!forum/souliss)

Italian Group, [here](https://groups.google.com/forum/#!forum/souliss-it)

Spanish Group, [here] (https://groups.google.com/forum/#!forum/souliss-es)

## Contribution

> Officiale repository for contribution in souliss github area: [here](https://github.com/souliss)

Reviewer: ```suggestion
Official repository for contributing to the Souliss project, GitHub page: [here](https://github.com/souliss)
```

## Download 

> To download latest compiled binding go to releases tab: [here](https://github.com/souliss/bindingopenhab2/releases)

Reviewer: Remove as once merged the binding is built into openHAB.

For updates go to official repository at [https://github.com/souliss/bindingopenhab2](https://github.com/souliss/bindingopenhab2)

> 

Reviewer: Can this be removed? As this is an opensource project, we only want to be promoting links to merged bindings that are community maintained.

# Souliss Binding

[Souliss](http://www.souliss.net/) is a networking framework for Arduino and compatibles boards, and is designed to let you easily build a smart home that is distributed over multiple boards via Ethernet, WiFi, wireless point-to-point and RS485 bus. 

> Souliss is an open-source and community driven project, you can use the [wiki](https://github.com/souliss/souliss/wiki) and [Community](https://github.com/souliss/souliss/wiki/Community) to get help and share you results.  

Reviewer: ```suggestion
Souliss is an open-source and community driven project, you can use the [wiki](https://github.com/souliss/souliss/wiki) and [Community](https://github.com/souliss/souliss/wiki/Community) to get help and share your results.  
```

## Prerequisites

> The binding requires a deployed network.  As a minimum, you need one Souliss node with Ethernet or WiFi access configured as a [Gateway](https://github.com/souliss/souliss/wiki/Gateway). A Gateway is a special node that is able to communicate with the user interfaces. The binding interacts as a user interface for Souliss.

Reviewer: Please put all sentence's on their own line.
This helps when lines change in the future.
Do this for all of the readme.

Author: Can you give me an example ? Not sure to understand it... One sentence for line ?

You can use SoulissApp and the Souliss binding at the same time, and generally up to five (by default, but can be increased) user interfaces simultaneously.

### Sketches

The easiest way is start with a simple example to control an ON/OFF light (though a relay). 
You can go to project [Souliss[(https://github.com/souliss/souliss), see a lot of examples sketches: [Souliss examples](https://github.com/souliss/souliss/tree/friariello/examples)

## Binding Configuration

This binding does not require any special configuration.

## Discovery

This binding can automatically discover devices. 
First Gateway Node, then Peer Nodes. 

## Supported Things

In Souliss Framework a Typical is one of predefined logic dedicated to smart home devices like lights, heating or antitheft. 

Typical can be one of T11, T12, T13, T14, etc... 

It are defined [here](https://github.com/souliss/souliss/wiki/Typicals).

Typicals match directly with openHAB Thing type.

|Device type |Typical Code | Thing type | 
|------------|---------|-------------------------------|

> |ON/OFF Digital Output with Timer Option|T11|souliss:t11|

Reviewer: My comment is about the 'Thing type' column and applies to them all. 
souliss:t11
in this case the thing type is 't11' and when doing textual config the part before the : refers to the binding name. Since it is not part of the 'thing type' name itself I see this as confusing and not how the other bindings present the data.

Author: renamed to simple thing type

|-- |-- | -- | -- | -- |--|--|--|--|--|--|--|--|--|--|--|--|--|--|--|--|
|souliss:t11|x|x|x|x||||
|souliss:t12|x| |x|x|x||||
|souliss:t13|||x|x||x|x||
|souliss:t14|||x|x||||x|
|souliss:t16|x|x|x|x|||||x|x|x|x||
|souliss:t18|x||x|x|||||||||
|souliss:t19|x|x|x|x||||||x|x|
|souliss:t1A|||x|x|||||||||x|x|x|x|x|x|x|x|

|Thing type / Channel | DateTime / lastStatusStored | Number / healthy |Rollershutter / rollershutter|(see down) / rollershutterstate|(see down) / mode|(see down) / fan|Switch / status|Number / setpoint|Switch / setAsMeasured|Switch / measured|Switch / statusAlarm|Switch / onOffAlarm|Switch / rearmAlarm|
|-- |-- | --|- | -- | -- |--|--|--|--|--|--|--|--|
|souliss:t21|x|x||x|
|souliss:t22|x|x|x|x|
|souliss:t31|x|x|||x|x|x|x|x|x|
|souliss:t41|x|x|||||||||x|x|x|
|souliss:t42|x|x|||||||||x||x|

rollershutterstate = opening, closing, limSwitch_open , limSwitch_close, state_open, state_close, NoLimSwitch

mode = COOLING_MODE, HEATING_MODE, POWEREDOFF_MODE

fan = AUTO, HIGH, MEDIUM, LOW, FANOFF

|Thing type / Channel| DateTime / lastStatusStored |Number / healthy|Number / value|
|-- |-- | -- | --|
|souliss:t51|x|x|x|

> |souliss:t52|x|x|x|

Reviewer: This table does not make anything clearer to me, I dont understand what it is trying to show?
Typically people use this section of the readme to learn what channels a thing type has and what they do and the type of channel is it so they know how to setup the thing with the correct type of controls.

Take a look at how other bindings display the data in a table and what headings they have.

Author: cutted souliss before :  
Normally who uses Souliss understand the table.

|souliss:t62|x|x|x|
|souliss:t63|x|x|x|
|souliss:t64|x|x|x|
|souliss:t65|x|x|x|
|souliss:t66|x|x|x|
|souliss:t67|x|x|x|
|souliss:t68|x|x|x|
|souliss:topic|x||x|

### Parameters

|Thing type| Parameters Name and Default Value| Description|
|-- |-- |-- |
|Gateway|gatewayIpAddress="" |Will be resolved by discovery if auto configured|
|"|gatewayPortNumber=230 ||
|"|preferredLocalPortNumber=0 |Default port is 0. It means that it is random|
|"|pingInterval=30 |Interval in seconds to check for device presence|
|"|subscriptionInterval=2|Interval in minutes to subcribe Souliss Gateway|
|"|healthyInterval=35|Interval in seconds to send nodes healthy|
|"|userIndex=70|Generally the default value work good. It must is different from other user interfaces (ex: SoulissApp)|
|"|nodeIndex=120|Generally the default value work good. It must is different from other user interfaces (ex: SoulissApp)|
|T11|sleep=5|Set sleep timer in cycles|
|T12|||
|T13|||
|T14|||
|T16|sleep=5|Set sleep timer in cycles|
|T19|sleep=5|Set sleep timer in cycles|
|T1A|||
|T21|||
|T22|||
|T31|||
|T4x|||
|T5x|||
|T6x|||

## Manual Things Configuration

If after discovery your thing is not listed you can add it manually.

> You have to choice it from disponible items. 

Reviewer: ```suggestion
You have to choose it from the available things. 
```

We try to keep generic how to use openHAB out of a bindings readme. This way if the UI changes we do not have a heaps of documents to update, as the information on how to add manual things is already covered once in the getting started guide. I feel some of this section can be removed to make it shorter for this reason.

Firts gateway, after items!

> To configure Gateway you can leave default value on Thing ID and write your value on "IP or Host Name" and "Gateway port".

Reviewer: I'll ignore the spelling mistake as I feel this is best removed for the reason above. Also you have these marked as required="true" in the XML file so the frame work will ensure they are filled in.

You can assume that a user has already read the getting started/ new user tutorial and knows how to use openHAB already. This also makes it less likely that the binding document needs to be updated when/if the openHAB UI changes how it works.

Author: You have the reason . Cutted

To configure a typical (items) you have to choice your "Name" and "Location", you have to choice your "Gateway" and insert correct "Thing ID".

Thing ID is [node]-[slot]
For example, if you have two nodes and you want configure a typical on second node at slot seven, you must write 
Thing ID: 

```
2-7
```

## Things  

Examples to configure items 
Thing <binding_id>:<type_id>:<thing_id> "Label" @ "Location" [ <parameters> ]
    
The general syntax for .things files is defined as follows (parts in <..> are required):

```

> Bridge <binding_id>:<type_id>:<bridge_id> "<Souliss Gateway Name>" [ <parameters> ]
> {  
> Thing <type_id> <thing_id>  [ <parameters> ]
> }

Reviewer: ```suggestion
Bridge <binding_id>:<bridge_thing_type_id>:<bridge_uid> "<Souliss Gateway Label>" [ <parameters> ]
{  
Thing <thing_type_id> <thing_uid> "<Souliss Thing Label>" [ <parameters> ]
}
```

Author: fixed as suggested
